File: 0179-largest-number/0179-largest-number.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function (nums) {


    nums.sort((a, b) => {
        let astring = String(a);
        let bstring = String(b);

        let poss1 = Number(astring + bstring); // 102
        let poss2 = Number(bstring + astring); // 210

        if(poss2> poss1){
            return 1;
        }
        else{
            return -1;
        }



    });
    let res="";
    for(let i=0;i<nums.length;i++){
        res += String(nums[i]);
    }
    while(res.length>1 && res[0]=='0'){
        res = res.slice(1);
    }
    return res;

};

File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool mycomparator(int a, int b) {
        string astr = to_string(a);
        string bstr = to_string(b);
        string p1 = astr + bstr;
        string p2 = bstr + astr;

        if (p1 > p2) {
            return true;
        } else {
            return false;
        }
    }
    string largestNumber(vector<int>& nums) {

        sort(nums.begin(), nums.end(), mycomparator);

        string res = "";
        for (int i = 0; i < nums.size(); i++) {
            res += to_string(nums[i]);
        }
        while(res.size()>1 && res[0]=='0'){
            res= res.substr(1);
        }
        return res;
    }
};